### ISSUE TITLE
starts_at doesn't behave as expected

### ISSUE THREAD

**Issue by emarchak** [Omerlo-Technologies/ex_cycle#19]
I'm using ex_cycle and when checking the occurrences of  a rule with a start date, the `starts_at` doesn't look like it's being included in the calculations. Am I using this wrong, or is this a bug?

Repo:
```elixir
    test "occurrences begin after starts_at date" do
      datetimes =
        ExCycle.new()
        |> ExCycle.add_rule(:daily, excluded_dates: [~D[2024-01-03]], starts_at: ~D[2024-01-02])
        |> ExCycle.occurrences(~D[2024-01-01])
        |> Enum.take(2)

      assert datetimes == [~N[2024-01-02 00:00:00], ~N[2024-01-04 00:00:00]]
    end
```

Results in the following failure:

```elixir
     Assertion with == failed
     code:  assert datetimes == [~N"2024-01-02 00:00:00", ~N"2024-01-04 00:00:00"]
     left:  [~N[2024-01-01 00:00:00], ~N[2024-01-02 00:00:00]]
     right: [~N[2024-01-02 00:00:00], ~N[2024-01-04 00:00:00]]
```

**Comment by Matsa59** [comment]
Hello, yeah I think the rule's `starts_at` must be used instead of the cycle's starts_at

**Comment by Matsa59** [comment]
BTW, thanks for the given test. Really appreciated.
I'm fixing it right now, you can expect a new version today

**Comment by Matsa59** [comment]
Fixed in 0.5.8 thanks for your contribution ;)

**Comment by emarchak** [comment]
No problem, thank you for the fix and release! It's working great.
